photonlib-java-examples/build.gradle

```diff
@@ -1,35 +1,145 @@
+                    exportedHeaders {
+                        srcDirs 'src/dev/native/include'
+                    }
+                }
+            }
+            binaries.all { binary ->
+                nativeUtils.useRequiredLibrary(binary, 'wpilib_executable_shared')
+                nativeUtils.useRequiredLibrary(binary, 'opencv_shared')
+                nativeUtils.useRequiredLibrary(binary, 'halsim_gui')
+            }
+        }
+    }
+    tasks {
+        def found = false
+        $.components.each {
+            if (it in NativeExecutableSpec && it.name == "dev") {
+                it.binaries.each {
+                    if (found)
+                        return
+
+                    def arch = it.targetPlatform.name
+                    if (arch == NativePlatforms.desktop) {
+                        found = true
+                        def filePath = it.tasks.install.installDirectory.get().toString() + File.separatorChar + 'lib'
+
+                        def doFirstTask = { task ->
+                            def extensions = ''
+                            it.tasks.install.installDirectory.get().getAsFile().eachFileRecurse {
+                                def name = it.name
+                                if (!(name.endsWith('.dll') || name.endsWith('.so') || name.endsWith('.dylib'))) {
+                                    return
+                                }
+                                def file = it
+                                if (name.startsWith("halsim_gui") || name.startsWith("libhalsim_gui".toString())) {
+                                    extensions += file.absolutePath + File.pathSeparator
+                                }
+                            }
+                            if (extensions != '') {
+                                task.environment 'HALSIM_EXTENSIONS', extensions
+                            }
+                        }
+
+                        new groovy.json.JsonSlurper().parseText(exampleFile.text).each { entry ->
+                            project.tasks.create("run${entry.foldername}", JavaExec) { run ->
+                                main = "org.photonlib.examples." + entry.foldername + ".Main"
+                                classpath = sourceSets.main.runtimeClasspath
+                                run.dependsOn it.tasks.install
+                                run.systemProperty 'java.library.path', filePath
+                                run.environment 'LD_LIBRARY_PATH', filePath
+                                run.workingDir filePath
+                                doFirst { doFirstTask(run) }
+
+                                if (org.gradle.internal.os.OperatingSystem.current().isMacOsX()) {
+                                    run.jvmArgs = ['-XstartOnFirstThread']
+                                }
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+}
```

Comment:
While I can't speak to the particular details of the implementation, the net effect is that it's functional on my windows PC.